snuba/reader.py

```diff
@@ -96,3 +98,30 @@ def execute(
             ),
             with_totals=with_totals,
         )
+
+
+class HTTPReader(Reader):
+    def __init__(
+        self, host: str, port: int, options: Optional[Mapping[str, str]] = None
+    ):
+        self.__base_url = f"http://{host}:{port}/"
+        self.__options = options if options is not None else {}
+
+    def execute(
+        self,
+        query: str,
+        settings: Optional[Mapping[str, str]] = None,
+        query_id: Optional[str] = None,
+        with_totals: bool = False,
+    ) -> Result:
+        parameters = {**self.__options}
+        if settings is not None:
+            parameters.update(settings)
+        if query_id is not None:
+            parameters["query_id"] = query_id
```

Comment:
This should probably ensure that it's not clobbering anything out of the default options mapping — the `HTTPReader` constructor should maybe disallow `query_id` as a parameter, as well as in the `settings` mapping as well?

These names should probably be made symmetric as well.